docs/user-guide/extending_argocd_with_plugins.md

```diff
@@ -0,0 +1,132 @@
+`argocd-my-plugin` binary in the PATH.
+
+## Limitations
+
+It is currently not possible to create plugins that overwrite existing
+`argocd` commands. For example, creating a plugin such as `argocd-version`
+will cause the plugin to never get executed, as the existing `argocd version`
+command will always take precedence over it. Due to this limitation, it is 
+also not possible to use plugins to add new subcommands to existing `argocd` commands.
+For example, adding a subcommand `argocd cluster upgrade` by naming your plugin 
+`argocd-cluster` will lead to the ignorance of plugin.
+
+## Conditions to follow for an `argocd` plugin
+
+Any binary that you would want to execute as an `argocd` plugin need to satisfy the following three conditions:
+
+1. The binary should start with `argocd-` as the prefix name. For example, `argocd-demo-plugin` is a valid binary but not `argocd_demo-plugin`.
+2. The binary should have executable permissions otherwise it will be ignored.
+3. The binary should reside anywhere in the system's PATH. However, plugins found in the relative path are ignored.
+
+## Writing `argocd` plugins
+
+A plugin can be written in any programming language or script that allows you to write command-line commands.
+
+A plugin determines which command path it wishes to implement based on its name.
+
+For example, If a binary named `argocd-demo-plugin` is available in your system's PATH, and the user runs the following command:
+
+```bash
+argocd demo-plugin subcommand1 --flag=true
+```
+
+Argo CD will translate and execute the corresponding plugin with the following command:
+
+```bash
+argocd-demo-plugin subcommand1 --flag=true
+```
+
+Similarly, if a plugin named `argocd-demo-demo-plugin` is found in the PATH, and the user invokes:
+
+```bash
+argocd demo-demo-plugin subcommand2 subcommand3 --flag=true
+```
+
+Argo CD will execute the plugin as:
+
+```bash
+argocd-demo-demo-plugin subcommand2 subcommand3 --flag=true
+```
+
+### Example plugin
+```bash
+#!/bin/bash
+
+# Check if the argocd CLI is installed
+if ! command -v argocd &> /dev/null; then
+    echo "Error: Argo CD CLI (argocd) is not installed. Please install it first."
+    exit 1
+fi
+
+if [[ "$1" == "version" ]]
+then
+    echo "displaying argocd version..."
+    argocd version
+    exit 0
+fi
+
+
+echo "I am a plugin named argocd-foo"
+```
+
+### Using a plugin
+
+To use a plugin, make the plugin executable:
+```bash
+sudo chmod +x ./argocd-foo
+```
+
+and place it anywhere in your `PATH`:
+```bash
+sudo mv ./argocd-foo /usr/local/bin
+```
+
+You may now invoke your plugin as a argocd command:
+```bash
+argocd foo
+```
+
+This would give the following output
+```bash
+I am a plugin named argocd-foo
+```
+
+All args and flags are passed as-is to the executable:
+```bash
+argocd foo version
+```
+
+This would give the following output
+```bash
+INFO[0000] command does not exist, looking for a plugin... 
```

Comment:
> INFO[0000] command does not exist, looking for a plugin... 

This first log line shouldn't be printed by the Argo CD CLI. Maybe if logs are running with DEBUG level but not as INFO

Reply:
implemented. Do you think we should print the warning too for a plugin that is not found in the PATH?
![image](https://github.com/user-attachments/assets/42e00264-7d64-4e45-bc47-b24488372434)